### PR TITLE
react-label: Updating spec, fixing typos in types, and adding descriptions in stories.

### PR DESCRIPTION
**Pull request by sopranopillow**
## PR Details

This PR adds the following:
- Updates the spec to reflect the implementation.
- Adds descriptions to stories.
- Adds note on an accessibility problem with contrast ratio not passing for disabled.
- Fixes a typo on types and changes `@defaultValue` to `@default`.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                 |         Size (minified/GZIP) |
| ------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>           | `29.382 kB`<br />`10.047 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr> |   `13.61 kB`<br />`5.462 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>        |  `25.312 kB`<br />`8.202 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/8437eb3b6023bdac5b234538e1116b1c7d24f53f'>8437eb3b6023bdac5b234538e1116b1c7d24f53f</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22833/perf-test-react-components/Avatar-mount_base.html">907</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22833/perf-test-react-components/Avatar-mount.html">889</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22833/perf-test-react-components/Button-mount_base.html">523</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22833/perf-test-react-components/Button-mount.html">546</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22833/perf-test-react-components/FluentProvider-mount_base.html">1910</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22833/perf-test-react-components/FluentProvider-mount.html">1863</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22833/perf-test-react-components/FluentProviderWithTheme-mount_base.html">276</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22833/perf-test-react-components/FluentProviderWithTheme-mount.html">269</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22833/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">244</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22833/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">229</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22833/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">320</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22833/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">323</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22833/perf-test-react-components/MakeStyles-mount_base.html">1573</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22833/perf-test-react-components/MakeStyles-mount.html">1550</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by tomi-msft — APPROVED** [review]

File: packages/react-components/react-label/src/stories/LabelStrong.stories.tsx [hunk 1]
```diff
@@ -2,3 +2,11 @@ import * as React from 'react';
 import { Label } from '../index'; // codesandbox-dependency: @fluentui/react-label ^9.0.0-beta
 
 export const Strong = () => <Label strong>Strong label</Label>;
+
+Strong.parameters = {
+  docs: {
+    description: {
+      story: 'A Label a semibold/strong fontweight.',
```
Comment (khmakoto) [inline packages/react-components/react-label/src/stories/LabelStrong.stories.tsx:9]: _nit:_
```suggestion
      story: 'A Label can have a semibold/strong fontweight.',
```

**Review by khmakoto — APPROVED** [review]

File: packages/react-components/react-label/src/stories/LabelSize.stories.tsx [hunk 2]
```diff
@@ -10,3 +10,11 @@ export const Size = () => {
     </>
   );
 };
+
+Size.parameters = {
+  docs: {
+    description: {
+      story: 'A Label supports `small`, `medium`, and `large` size.',
```
Comment (spmonahan) [inline packages/react-components/react-label/src/stories/LabelSize.stories.tsx:17]:
```suggestion
      story: 'A Label supports `small`, `medium`, and `large` sizes.',
```

File: packages/react-components/react-label/src/stories/LabelStrong.stories.tsx (hunk 1 above)
Comment (spmonahan) [inline packages/react-components/react-label/src/stories/LabelStrong.stories.tsx:9]:
```suggestion
      story: 'A Label with a strong font weight.',
```

**Review by spmonahan — APPROVED** [review]

**Pull request merged**